### PR TITLE
Support non Swift files in `--file` argument

### LINKED ISSUE #1721: Swiftlint only expects files with type ".swift"
I know that swiftlint explicitly expects files of type .swift, but I'd like to pass a file with ".lint-test" instead of ".swift". This is mainly for easy integration with https://github.com/phacility/arcanist.

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes #1721.

### CODE REVIEW

File: Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift [hunk 1]
```diff
@@ -21,7 +21,7 @@ extension FileManager: LintableFileManager {
             .standardizingPath
 
         // if path is a file, it won't be returned in `enumerator(atPath:)`
-        if absolutePath.bridge().isSwiftFile() && absolutePath.isFile {
+        if absolutePath.isFile {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: why this change? I think the `lintablePaths` change is sufficient to support the `lint --path file.notswift` scenario.

Comment (marcelofabri) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: I guess that's true, but when exactly would this check be true then? 🤔

Comment (jpsim) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: when specifying `included` and `excluded` configuration values, which IMO doesn't really make sense to specify non-Swift files in there.

Comment (jpsim) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: I'm kinda worried that if we extend `included`/`excluded` to support globbing, this change would attempt to lint non-Swift files if `Sources/**` were specified, for example.

Comment (marcelofabri) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: I've removed it 👍
